app/views/flush.py

```diff
@@ -0,0 +1,66 @@
+from app.globals import get_answer_store, get_metadata, get_questionnaire_store
+from app.submitter.submitter import SubmitterFactory
+from app.submitter.converter import convert_answers
+from app.utilities.schema import get_schema
+
+from app.authentication.permission_denied_exception import PermissionDeniedException
+from app.authentication.survey_data_exception import SurveyDataException
+from app.authentication.no_token_exception import NoTokenException
+from app.authentication.user import User
+from app.authentication.user_id_generator import UserIDGenerator
+from app.authentication.authenticator import jwt_decrypt
+
+from app.questionnaire.path_finder import PathFinder
+
+from flask import Blueprint, Response, request, session
+
+flush_blueprint = Blueprint('flush', __name__)
+
+
+@flush_blueprint.route('/flush', methods=['GET'])
+def flush_data():
+
+    if session:
+        session.clear()
+
+    encrypted_token = request.args.get('token')
+
+    if encrypted_token is None:
+        raise NoTokenException("Please provide a token")
+
+    decrypted_token = jwt_decrypt(encrypted_token)
+
+    roles = decrypted_token.get('roles')
+
+    if roles:
+        if 'flusher' in roles:
+            user = _get_user(decrypted_token)
+            _submit_data(user)
+            get_questionnaire_store(user.user_id, user.user_ik).delete()
+        else:
+            raise PermissionDeniedException("Your role doesn't have permission to flush data")
+    else:
+        raise PermissionDeniedException("No role defined, you can't flush data")
+
+    return Response(status=200)
+
+
+def _submit_data(user):
+
+    answer_store = get_answer_store(user)
+
+    if answer_store.answers:
+        metadata = get_metadata(user)
+        schema = get_schema(metadata)
+        routing_path = PathFinder(schema, answer_store, metadata).get_routing_path()
+        submitter = SubmitterFactory.get_submitter()
+        message = convert_answers(metadata, schema, answer_store, routing_path)
+        submitter.send_answers(message)
+    else:
+        raise SurveyDataException("There are no answers to flush")
```

Comment:
Do we want to raise a SurveyDataException here, this will cause us to return a 404 web page which doesn't seem right for an API designed for a system